test_pull_requests

```diff
@@ -935,6 +961,86 @@ popd
+        comment_matches_syntax = comment['body'] =~ /#{Regexp.quote(build_url)}/ && comment['body'] =~ general_issue_spec
+        # we can't allow the bot to be a valid author, or we would pick up
+        # comments where the bot explains the flake syntax to GitHub users
+        comment_author_valid = comment['user']['login'] != Properties['bot_github_user']
+        if comment_matches_syntax && comment_author_valid
+          # if we find a comment that matches the flake job syntax from someone
+          # trusted, we need to validate the comment's flake issue list
+          bad_issue_links = []
+          comment['body'].scan(general_issue_spec) do |issue|
+            if issue =~ flake_issue_spec
+              issue_url = $1
+              issue_id = $2.to_i
+              if !issue_has_test_flake_label(issue_id, settings['flake_config']['flake_repo'], settings['flake_config']['flake_label'])
+                # the issue does not have the labels necessary, so it is invalid
+                bad_issue_links << issue_url
+              end
+            else
+              # we had a GitHub issue link, but it didn't 
+              # match the org or repo that we wanted
+              bad_issue_links << issue
+            end
+          end
+
+          if bad_issue_links.length == 0
+            valid_comment_exists = true
+            break
+          else
+            # any invalid issue links mean that the comment is invalid
+            reason = %{A [comment](#{comment["html_url"]}) was found that referenced the correct Jenkins job URL, but some listed issue links were invalid. Ensure all linked issues are in the [`#{Properties['github_user']}/#{settings['flake_config']['flake_repo']}` repository](https://github.com/#{Properties['github_user']}/#{settings['flake_config']['flake_repo']}/issues) and have the `kind/test-flake` label. Invalid links are:
+ - #{bad_issue_links.join("\n - ")}}
+          end
+        end
+      end
+
+      [valid_comment_exists, reason]
+    end
+
+    # issue_has_test_flake_labels determines if the issue at the
+    # repository and issue ID has a test flake label
+    def issue_has_test_flake_label(issue_id, repo, flake_label)
+      get_labels(issue_id, repo).each do |label|
+        if label["name"] == flake_label
+          return true
+        end
+      end
+      return false
+    end
+
+    # format_flake_comment builds a comment explaining why a re-build could not be triggered
+    def format_flake_comment(repo, job, reason)
+      %{#{flake_denied_prefix(repo, job)}
+In order for the @#{Properties['bot_github_user']} to run the #{Properties['repo_to_product'][repo]} #{job} job again, one of the following actions is necessary:
+ - A contributor needs to update this pull request, *e.g.* to remove a failure cause it introduced
+ - A privileged contributor needs to trigger the job
+ - A contributor needs to leave a comment linking the failure to test or infrastructure flake issues
```

Comment:
Contributor is not a reserved keyword, as in we allow anyone to contribute but not all contributors are trusted, etc. I want to keep this since it makes the three bullet points parallel and keeps me sane